gui_agents/service/agent_service.py

```
@@ -259,22 +281,17 @@ def _execute_task_internal(self, request: TaskRequest, task_result: TaskResult)
-                    del Registry._services[registry_key]
-            except:
-                pass
+            # Cleanup task-specific registry
+            Registry.remove_task_registry(task_result.task_id)
+            self.logger.info(f"Cleaned up task-specific registry for task {task_result.task_id}")
         
         return task_result
     
-    def _run_agent_normal_internal(self, agent, instruction: str, hwi, max_steps: int, 
+    def _run_agent_normal_internal(self, agent, instruction: str, hwi, max_steps: int,
                                  enable_takeover: bool, task_id: str):
```

Ruff (0.14.0)

Unused method argument: `enable_takeover`

(ARG002)